### PR TITLE
[refactor](fs)(step2)separate the storage and filesystem methods

### PR DESCRIPTION
**Pull request by wsjz**
# Proposed changes

see https://github.com/apache/doris/issues/18960

PR1: add new storage file system template and move old storage to new package
PR2: extract some method in old storage to new file system.
PR3: use storages to access remote object storage, and use file systems to access file in local or remote location. Will add some unit tests.

this is PR2.

adjust the diagrams:
image

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/S3Resource.java [hunk 1]
```diff
@@ -128,17 +128,17 @@ private static boolean pingS3(CloudCredentialWithEndpoint credential, String buc
         propertiesPing.put(S3Properties.Env.REGION, credential.getRegion());
         propertiesPing.put(PropertyConverter.USE_PATH_STYLE, "false");
         properties.putAll(propertiesPing);
-        S3Storage storage = new S3Storage(properties);
+        S3FileSystem fileSystem = new S3FileSystem(properties);
```
Comment (wsjz) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/S3Resource.java:131]: Test is pass

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/S3TableValuedFunction.java [hunk 2]
```diff
@@ -89,17 +88,12 @@ public S3TableValuedFunction(Map<String, String> params) throws AnalysisExceptio
         locationProperties.put(PropertyConverter.USE_PATH_STYLE, usePathStyle);
         if (FeConstants.runningUnitTest) {
             // Just check
-            BlobStorage.create(null, StorageBackend.StorageType.S3, locationProperties);
+            FileSystemFactory.get(StorageBackend.StorageType.S3, locationProperties);
```
Comment (wsjz) [inline fe/fe-core/src/main/java/org/apache/doris/tablefunction/S3TableValuedFunction.java:91]: just for UT, so can use FileSystem here.

File: fe/fe-core/src/main/java/org/apache/doris/fs/PersistentFileSystem.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import org.apache.doris.common.io.Writable;
+
+import com.google.common.collect.Maps;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Map;
+
+public abstract class PersistentFileSystem implements FileSystem, Writable {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/fs/PersistentFileSystem.java:31]: Add comment

**Review by morningman — COMMENTED** [review]
Please add comment for major interfaces or classes

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**